ssz/abc.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Iterable, Iterator, TypeVar, Union
+
+from eth_typing import Hash32
+from pyrsistent.typing import PVector
+
+from ssz.hash_tree import HashTree
+
+TStructure = TypeVar("TStructure")
+TElement = TypeVar("TElement")
+
+
+class HashableStructureAPI(ABC, Generic[TElement]):
```

Comment:
I don't think so, because `HashableStructureAPI` is the base class for all hashable structures, so the subclasses already "know" what type they are, so we don't need a type variable for it. `TStructure` is used in some methods of the class, but always bound to `self`, so we don't need to bind it more globally.

[In general, the type annotations in py-ssz are really broken though, and not checked in CI, so don't trust them too much]